packages/components/src/spectrum/CheckboxGroup.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+  Checkbox,
+  CheckboxGroup as SpectrumCheckboxGroup,
+  SpectrumCheckboxGroupProps,
+} from '@adobe/react-spectrum';
+import React, { useMemo } from 'react';
+
+/**
+ * Augmented version of the Spectrum CheckboxGroup component that supports
+ * primitive item children.
+ */
+export function CheckboxGroup({
+  children,
+  ...props
+}: SpectrumCheckboxGroupProps): JSX.Element {
+  const wrappedChildren = useMemo(
+    () =>
+      React.Children.map(children, child => {
+        if (typeof child === 'function') {
+          return child;
+        }
+        return <Checkbox>{child}</Checkbox>;
```

Comment:
This has issues. I tried this in the styleguide:

```typescript
<CheckboxGroup>
       <SpectrumCheckbox>Aaa</SpectrumCheckbox>
</CheckboxGroup>
```

And it shows a nested checkbox:
<img width="128" alt="image" src="https://github.com/user-attachments/assets/07fa0a52-dc2b-45ac-8627-22b4aed66058">

Are there actually cases where the child would be a `function`?

I would recommend putting some examples in the styleguide to prove the different child types work (I did this with ListView and Picker)